clippy_utils/src/usage.rs

```diff
@@ -37,6 +37,17 @@ pub fn is_potentially_mutated<'tcx>(variable: HirId, expr: &'tcx Expr<'_>, cx: &
     mutated_variables(expr, cx).map_or(true, |mutated| mutated.contains(&variable))
 }
 
+pub fn is_potentially_local_place(local_id: HirId, place: &Place<'_>) -> bool {
+    match place.base {
+        PlaceBase::Local(id) => id == local_id,
+        PlaceBase::Upvar(_) => {
+            // Conservatively assume yes.
+            true
+        },
+        _ => false,
+    }
+}
+
```

Comment:
Seems like this wouldn't support static variables? Could you add a test for `static` places?

Reply:
Wdym by this? If the passed place is a static variable, then it sounds right that this should return false, since it's not a local place. It's ok for this lint specifically to only consider locals and ignore statics, because it doesn't/didn't look for statics anyway. That is, this already doesn't lint:
```rs
fn f() {
  static X: Option<i32> = Some(123);

  if X.is_some() {
    X.unwrap();
  }
}
```
so there's no way to really get FPs here if this returns `false` for a static place there I think.
I suppose it would still be worth having a test that makes sure that something like
```rs
static mut X: Option<i32> = Some(123);
if X.is_some() {
  X = None;
  X.unwrap();
}
```
doesn't lint

Reply:
It would be nice if we could also lint static places, although calling unwrap on an static variable isn't common (nor a good practice). Why are not linting on static places, just because of backwards compatbility?

Reply:
I'm not sure why it doesn't lint on those, it doesn't mention that anywhere in the lint as far as I can see. Maybe it was just forgotten as it's probably very uncommon to do this on statics like you said. I think it would be too big of a change for this PR to make that work imo. There seem to be quite a few places in this lint that assume it's a local